### ISSUE TITLE
Docs are Inconsistent for VirtualNode backends array

### ISSUE THREAD

**Issue by kferrone** [aws/aws-app-mesh-controller-for-k8s#175]
So I am new to AppMesh. I have our apps working with AppMesh and so far so good. I simply added the CRD kinds in our Helm charts. However I did run into some issues with the backend services. 

I am also having trouble finding some more detailed examples of using the CRDs in k8s. I read through the CRDs and it seems inconsistent with [the docs I read here](https://docs.aws.amazon.com/app-mesh/latest/userguide/mesh-k8s-integration.html) and the [docs here](https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/docs/design.md).  I also can't find any examples of using Appmesh CRDs in the official examples repo either, [aws/aws-app-mesh-examples](https://github.com/aws/aws-app-mesh-examples), it is always using cloudformation. 

The main issue is the docs say there is a backend section like below for `VirtualNodes`:
```yaml
backends:
  - virtualService:
       virtualServiceName: my-service-b.my-app-1.svc.cluster.local
```
However in the VirtualNode CRD, I can't find any mention of `virtualService` or `virtualServiceName` and it's definitely not in the backends array. I do see a definition for backends array like below:
```yaml
oneOf:
- type: object
   properties:
     backendService:
       type: object
       properties:
         clientPolicy:
           type: object
           properties: ...
         name:
           type: string  
```
which suggests the examples and docs should be like below:
```yaml
backends:
  - backendService:
       name: my-service-b.my-app-1.svc.cluster.local
```
Am I correct? Should I trust the docs or the CRD?

**Comment by lavignes** [comment]
Hey @kferrone 
That does seem off. We do have examples that use CRDs in our examples repo. They are under the `walkthroughs/howto-k8s-*` directories. eg: https://github.com/aws/aws-app-mesh-examples/tree/master/walkthroughs/howto-http-headers

It looks like someone else also noticed this inconsistency: https://github.com/aws/aws-app-mesh-controller-for-k8s/issues/140

You should trust the docs. We should definitely fix the CRDs.

**Comment by kferrone** [comment]
Thank you, I have found the examples I was looking for. I ended up learning the hard way the docs are to be trusted, following the CRDs just made bugs happen.

**Comment by M00nF1sh** [comment]
@kferrone 
if you are learning, please according to our new v1beta2 API: https://github.com/aws/aws-app-mesh-controller-for-k8s/pull/150, it address all these semantic and buggy behavior in v1beta1 API.

It's already code ready, and we'll release it soon. 
You can find the API types here: https://github.com/aws/aws-app-mesh-controller-for-k8s/tree/v1beta2/apis/appmesh/v1beta2
And CRD definitions here: https://github.com/aws/aws-app-mesh-controller-for-k8s/tree/v1beta2/config/crd/bases

If you want to try it out before we release it:
1. delete all CR/CRD and old appmesh controller/injector
2. checkout the repo and checkout v1beta2 branch.
3. `export IMG=<your-dockerhub-repo>:<someTag>`
4.  `make docker-build`
5. `make docker-push`
6. `make deploy`

**Comment by astrived** [comment]
Resolved in v1.0.0 release.
